src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -68,22 +70,35 @@ private async Task<Solution> MakeStaticAsync(Document document, SyntaxNode root,
             var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
             if (symbol != null)
             {
-                solution = await UpdateReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+                var (newSolution, allReferencesFixed) = await UpdateReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+                solution = newSolution;
+
+                if (!allReferencesFixed)
+                {
+                    // We could not fix all references, so add a warning annotation that users need to manually fix these.
+                    document = await AddWarningAnnotation(solution.GetDocument(document.Id), symbol, cancellationToken).ConfigureAwait(false);
+                    solution = document.Project.Solution;
+                }
             }
 
             return solution;
         }
 
-        private async Task<Solution> UpdateReferencesAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
+        /// <summary>
+        /// Returns the updated solution and a flag indicating if all references were fixed or not.
+        /// </summary>
+        private async Task<(Solution newSolution, bool allReferencesFixed)> UpdateReferencesAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
             var references = await SymbolFinder.FindReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
             if (references.Count() != 1)
             {
-                return solution;
+                return (newSolution: solution, allReferencesFixed: true);
```

Comment:
Shouldn't this still be false when references.Count() > 1?

Reply:
Thanks! Fixing that uncovered that the current implementation did not work for properties, as find all references returned references for cascaded accessors, and we were bailing out. I have fixed it along with unit tests in the latest commit.

Reply:
Seems another argument for https://github.com/dotnet/roslyn/issues/25628